encfs/CipherFileIO.cpp

```diff
@@ -535,7 +535,7 @@ ssize_t CipherFileIO::read(const IORequest &origReq) const {
    * to the data. */
   if (req.offset < 0) {
     headerBytes = -req.offset;
-    if (req.dataLen < headerBytes) {
+    if (req.dataLen < (size_t)headerBytes) {
```

Comment:
Shouldn't headerBytes be unsigned?

Reply:
Yep, but HEADER_SIZE itself is a int...
Moving headerBytes to unsigned int would then lead to some limit considerations, for example line 545.
You point to #447 :)

Reply:
I had an additional look, if we make HEADER_SIZE unsigned, other casts will be needed as it is compared to st_size which is an off_t...

Reply:
Hmm. Maybe we should move everything to ssize_t or int64_t.

Reply:
I also thought about it, but I think `dataLen` should remain `size_t`, as in `read(2)` and `write(2)`.